### PR TITLE
Allow selection of different audio tracks for tv shows

### LINKED ISSUE #478: Add audio track selection option
Unable to choose audio track for media which has multiple audio tracks for different languages.

### PR DESCRIPTION
**Pull request by jimdogx**
Allow selection of different audio tracks for TV shows.

When selecting "option" on an episode with more than one audio track, the user can now select a different audio track.

This PR will also add text ("+1", "+2", etc) to the audio codec if more than one audio track is found.  Because of this, I've noticed that loading the rows takes a smidge longer.  So the Season Title will now say "Loading..." until the rows are done.

**Issues**
Fixes: #478 

**Known Issues**
If you play an episode, then go back and select a different audio track, the episode (with newly selected audio track) begins to play immediately.  This could be viewed as a feature 😄   Can be looked at in a future release.

### CODE REVIEW

**Comment by neilsb** [comment]
Looks good.  I'll try to give it a proper test over the next few days.  👍

**Comment by neilsb** [comment]
Couple of minor things I found during functional testing

 - From the episode list, if you press * to open the options, then press * again (I assumed that's how to close them) focus gets lost and there doesn't appear to be a way to get it back
 - If you start playing an episode, then hit "Back" to stop and return to then episode list, press * and choose a different track then back out of the options, the episode starts playing again automatically. Doesn't do it if you don't select a different track. Not saying this is necessarily an issue, just maybe a little odd?

**Comment by jimdogx** [comment]
Thanks!

I did run in to the issue where the RowList loses focus, but I added some logic so that if the user hits "up" or "down" it regains focus.  However, I don't think I ever tried hitting "*" to go back (only "back").  So you might have found another case.

I'll take a look at that and the un-intended autoplay when I get a chance 👍

**Comment by jimdogx** [comment]
I think I fixed the "*" issue.

Will have to spend some time looking in to the autoplay issue later.  Although it is a nice "feature" wouldn't you say? 😄

**Comment by neilsb** [comment]
> Will have to spend some time looking in to the autoplay issue later. Although it is a nice "feature" wouldn't you say? 😄

Genuinely I don't think it's bad.  If someone is going into that menu and selects an audio track, it's logical that they want to play it.  Making it autoplay if they choose one on a non-started episode would be just as valid since it would be consistent  😄

File: source/Main.brs [hunk 1]
```diff
@@ -103,7 +107,11 @@ sub Main (args as dynamic) as void
                 ' play episode
                 ' todo: create an episode page to link here
                 video_id = selectedItem.id
-                video = CreateVideoPlayerGroup(video_id)
+                if selectedItem.selectedAudioStreamIndex > 1
```
Comment (neilsb) [inline source/Main.brs]:
```suggestion
                if selectedItem.selectedAudioStreamIndex <> invalid and selectedItem.selectedAudioStreamIndex > 1
```

Trying to play a TV Episode directly from the home screen (in Next up, for example) causes a crash here, as `selectedItem.selectedAudioStreamIndex` is not set

**Review by neilsb — CHANGES_REQUESTED** [review]
If you have an episode that is part way played, and go into the TV Show and Season, and select it from the episode list then you get the popup about starting over or resuming.   If you select one the episode starts.  If you then press back you end up back on the episode list, but none of the episodes have focus.   Pressing * at this point causes a crash.

I realise the root cause of this is that focus is not set properly after the Resume/Restart appears, but probably worth catching this.

File: source/Main.brs (hunk 1 above)
Comment (jimdogx) [reply source/Main.brs]: Thanks... Spent another hour working on the "auto play" issue and as part of that I had added invalid checks.  However, I did not test the resume/restart then back flow; so great catch!

**Review by neilsb — APPROVED** [review]
All looks good 👍

**Pull request merged**